test/test_foreach.py

```diff
@@ -900,6 +900,31 @@ def test_outplace_with_invalid_grads(self, device, dtype, op):
         self.assertIsNotNone(sample.input[0].grad)
         self.assertIsNone(sample.input[1].grad)
 
+    # note(crcrpar): this is clumsy but we don't have out-place `_foreach_zero` so make it a special case.
```

Comment:
Could we make an OpInfo for the inplace version? (we have an OpInfo for the inplace `zero_`)

Reply:
Yes we could but I thought this would be simpler (sorry for being lazy haha)

Some changes would be needed in `ForeachFuncInfo`. Currently ForeachFuncInfo I wrote previously assumes that out-place is always exists while in-place not. This restriction derives from around
https://github.com/pytorch/pytorch/blob/49c8a0cad0dc314fba05c0a7d909df0b2ab991ab/torch/testing/_internal/opinfo/core.py#L2634-L2644

Reply:
Hmm I'm still concerned about how this makes it harder to test things consistently across the the ops.

I'd be okay with a special case in `get_foreach_method_names`, e.g:
```
if name == "zero_":
    <do something else>
```